extensions/cms/src/cmsUtils.ts

```diff
@@ -24,9 +31,35 @@ const mssqlProvider: string = 'MSSQL';
  */
 export class CmsUtils {
 
+	private _credentialProvider: azdata.CredentialProvider;
 	private _cmsService: mssql.CmsService;
 	private _registeredCmsServers: ICmsResourceNodeInfo[];
 
+	constructor() {
+		// 1) Get a credential provider
+		// 2) Store the credential provider for use later
+		this.setupCredentialProvider();
+	}
+
+	private async setupCredentialProvider(): Promise<azdata.CredentialProvider> {
+		if (!this._credentialProvider) {
+			this._credentialProvider = await azdata.credentials.getProvider(CredentialNamespace);
+		}
```

Comment:
This is still duplicating the logic in credentialProvider() below - I'd say just call that method in the constructor and get rid of this one. But be careful since that still has the possible race condition I've outlined before if the constructor is called and then something like savePassword is called before the first call to credentialProvider() finishes.